test/common/router/config_impl_test.cc

```diff
@@ -2031,6 +2036,228 @@ TEST(RoutePropertyTest, excludeVHRateLimits) {
   EXPECT_TRUE(config_ptr->route(headers, 0)->routeEntry()->includeVirtualHostRateLimits());
 }
 
+TEST(CustomRequestHeadersTest, AddNewHeader) {
+  const std::string json = R"EOF(
+	{
+  "virtual_hosts": [
+    {
+      "name": "www2",
+      "domains": [
+        "lyft.com",
+        "www.lyft.com",
+        "w.lyft.com",
+        "ww.lyft.com",
+        "wwww.lyft.com"
+      ],
+      "request_headers_to_add": [
+        {
+          "key": "x-client-ip",
+          "value": "%CLIENT_IP%"
+        }
+      ],
+      "routes": [
+        {
+          "prefix": "/new_endpoint",
+          "prefix_rewrite": "/api/new_endpoint",
+          "cluster": "www2",
+          "request_headers_to_add": [
+            {
+              "key": "x-client-ip",
+              "value": "%CLIENT_IP%"
+            }
+          ]
+        },
+        {
+          "path": "/",
+          "cluster": "root_www2",
+          "request_headers_to_add": [
+            {
+              "key": "x-client-ip",
+              "value": "%CLIENT_IP%"
+            }
+          ]
+        },
+        {
+          "prefix": "/",
+          "cluster": "www2"
+        }
+      ]
+    },
+    {
+      "name": "www2_staging",
+      "domains": [
+        "www-staging.lyft.net",
+        "www-staging-orca.lyft.com"
+      ],
+      "request_headers_to_add": [
+        {
+          "key": "x-vhost-header1",
+          "value": "vhost1-www2_staging"
+        }
+      ],
+      "routes": [
+        {
+          "prefix": "/",
+          "cluster": "www2_staging",
+          "request_headers_to_add": [
+            {
+              "key": "x-client-ip",
+              "value": "%CLIENT_IP%"
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "name": "default",
+      "domains": [
+        "*"
+      ],
+      "routes": [
+        {
+          "prefix": "/",
+          "cluster": "instant-server",
+          "timeout_ms": 30000
+        }
+      ]
+    }
+  ],
+  "request_headers_to_add": [
+    {
+      "key": "x-client-ip",
+      "value": "%CLIENT_IP%"
+    }
+  ]
+}
+	  )EOF";
+  NiceMock<Runtime::MockLoader> runtime;
+  NiceMock<Upstream::MockClusterManager> cm;
+  NiceMock<Envoy::Http::AccessLog::MockRequestInfo> requestInfo;
+  ConfigImpl config(parseRouteConfigurationFromJson(json), runtime, cm, true);
+
+  // Request header manipulation testing.
+  {
+    {
+      std::string s1 = ("127.0.0.1");
+      Http::TestHeaderMapImpl headers = genHeaders("www.lyft.com", "/new_endpoint/foo", "GET");
+      ON_CALL(requestInfo, getDownstreamAddress()).WillByDefault(ReturnRef(s1));
+      const RouteEntry* route = config.route(headers, 0)->routeEntry();
+      route->finalizeRequestHeaders(headers, requestInfo);
+      EXPECT_EQ("127.0.0.1", headers.get_("x-client-ip"));
+    }
+  }
+}
+
+bool CheckFormatMessage(const std::string errorMessage) {
```

Comment:
nit: `static bool CheckFormatMessage`, use snake case for param (please check everywhere else for camelCase that should be snake_case)